test/widgets/theme_test.dart

```diff
@@ -76,30 +76,71 @@ void main() {
+      });
+
+      testWidgets('dark -> light', (tester) async {
+        final a = DesignVariables.dark();
+        final b = DesignVariables.light();
+        check(() => a.lerp(b, 0.5)).returnsNormally();
+      });
+
+      testWidgets('dark -> dark', (tester) async {
+        final a = DesignVariables.dark();
+        final b = DesignVariables.dark();
+        check(() => a.lerp(b, 0.5)).returnsNormally();
+      });
+    });
+  });
+
   group('colorSwatchFor', () {
-    void doTest(String description, int baseColor, StreamColorSwatch expected) {
-      testWidgets('$description $baseColor', (WidgetTester tester) async {
-        addTearDown(testBinding.reset);
+    const baseColor = 0xff76ce90;
 
-        final subscription = eg.subscription(eg.stream(), color: baseColor);
+    testWidgets('light–dark animation', (WidgetTester tester) async {
+      addTearDown(testBinding.reset);
 
-        await tester.pumpWidget(const ZulipApp());
-        await tester.pump();
+      final subscription = eg.subscription(eg.stream(), color: baseColor);
 
-        late StreamColorSwatch actualSwatch;
-        final navigator = await ZulipApp.navigator;
-        navigator.push(MaterialWidgetRoute(page: Builder(builder: (context) {
-          actualSwatch = colorSwatchFor(context, subscription);
-          return const Placeholder();
-        })));
-        await tester.pumpAndSettle();
+      debugFollowPlatformBrightness = true;
+      tester.platformDispatcher.platformBrightnessTestValue = Brightness.light;
+      addTearDown(tester.platformDispatcher.clearPlatformBrightnessTestValue);
 
-        // Compares all the swatch's members; see [ColorSwatch]'s `operator ==`.
-        check(actualSwatch).equals(expected);
-      });
-    }
+      await tester.pumpWidget(const ZulipApp());
+      await tester.pump();
+
+      final navigator = await ZulipApp.navigator;
+      navigator.push(MaterialWidgetRoute(page: Builder(builder: (context) =>
+        const Placeholder())));
+      await tester.pumpAndSettle();
+
+      final element = tester.element(find.byType(Placeholder));
+      // Compares all the swatch's members; see [ColorSwatch]'s `operator ==`.
+      check(colorSwatchFor(element, subscription))
+        .equals(StreamColorSwatch.light(baseColor));
+
+      tester.platformDispatcher.platformBrightnessTestValue = Brightness.dark;
+      await tester.pump();
+
+      await tester.pump(kThemeAnimationDuration * 0.4);
+      final expectedLerped = StreamColorSwatch.lerp(
+        StreamColorSwatch.light(baseColor),
+        StreamColorSwatch.dark(baseColor),
+        0.4)!;
+      check(colorSwatchFor(element, subscription)).equals(expectedLerped);
 
-    doTest('light', 0xff76ce90, StreamColorSwatch.light(0xff76ce90));
-    // TODO(#95) test with Brightness.dark and lerping between light/dark
+      await tester.pump(kThemeAnimationDuration * 0.6);
+      check(colorSwatchFor(element, subscription)).equals(StreamColorSwatch.dark(baseColor));
```

Comment:
nit: wrap line, also making parallel to the light case above
```suggestion
      check(colorSwatchFor(element, subscription))
        .equals(StreamColorSwatch.dark(baseColor));
```

Then might as well make the intermediate case parallel too: inline `expectedLerped`, and format the same way, like
```dart
      check(colorSwatchFor(element, subscription))
        .equals(StreamColorSwatch.lerp(
```